internal/gengapic/gengapic.go

```diff
@@ -619,11 +619,25 @@ func (g *generator) insertMetadata(m *descriptor.MethodDescriptorProto) error {
 			}
 			seen[field] = true
 
+			accessor := fmt.Sprintf("req%s", buildAccessor(field))
+			typ := g.lookupFieldType(m.GetInputType(), field)
+
+			// TODO(noahdietz): need to handle []byte for TYPE_BYTES.
+			if typ == descriptor.FieldDescriptorProto_TYPE_STRING {
+				accessor = fmt.Sprintf("url.QueryEscape(%s)", accessor)
+			} else if typ == descriptor.FieldDescriptorProto_TYPE_DOUBLE || typ == descriptor.FieldDescriptorProto_TYPE_FLOAT {
+				// Format the floating point value with mode 'g' to allow for
+				// exponent formatting when necessary, and decimal when adequate.
+				// QueryEscape the resulting string in case there is a '+' in the
+				// exponent.
+				// See golang.org/pkg/fmt for more information on formatting.
+				accessor = fmt.Sprintf("url.QueryEscape(fmt.Sprintf(\"%%g\", %s))", accessor)
```

Comment:
Consider using \` to avoid having to escape the double quotes.

Reply:
Done